clang/lib/Sema/SemaDeclAttr.cpp

```diff
@@ -4489,6 +4489,21 @@ static void handleSYCLDeviceIndirectlyCallableAttr(Sema &S, Decl *D,
   handleSimpleAttribute<SYCLDeviceIndirectlyCallableAttr>(S, D, AL);
 }
 
+static void handleSYCLRegisterNumAttr(Sema &S, Decl *D, const ParsedAttr &AL) {
+  auto *VD = cast<VarDecl>(D);
+  if (!VD->hasGlobalStorage()) {
+    S.Diag(AL.getLoc(), diag::err_sycl_attibute_cannot_be_applied_here)
+        << AL << 0;
+    return;
+  }
+  assert(AL.getNumArgs() == 1);
```

Comment:
Do you have a test to make sure that this assert isn't hit when providing more args?  A number of functions here check to make sure they have the right number of args.

Reply:
I don't have a test. Is there a policy that each assert should have a test?

Reply:
Well, you're doing something significantly different from what every other diagnostic happens.  I presumed you thought the assert was safe (rather than checking that you have the correct number of args like everyone else here), so I wanted you to confirm it with a test.

Reply:
OK, I see your point now. I will change the assert to compiler check with diagnostics and add a negative test.